lib/models/chat_history.dart

```diff
@@ -223,6 +224,32 @@ Stream<DeltaEvent> _handleDocumentChange(
           pinned: false);
       yield AppendDeltaEvent(model);
       break;
+    case "channel.channel_points_custom_reward_redemption.add":
+      final status = data['event']['status'];
+      if (status == "fulfilled") {
+        final model = TwitchChannelPointRedemptionEventModel(
+            pinned: false,
+            messageId: change.doc.id,
+            redeemerUsername: data['event']['user_name'],
+            status: status,
+            rewardName: data['event']['reward']['title'],
+            rewardCost: data['event']['reward']['cost']);
+        yield AppendDeltaEvent(model);
+      }
+      break;
+    case "channel.channel_points_custom_reward_redemption.update":
+      final status = data['event']['status'];
+      if (status == "fulfilled") {
+        final model = TwitchChannelPointRedemptionEventModel(
+            pinned: false,
+            messageId: change.doc.id,
+            redeemerUsername: data['event']['user_name'],
+            status: status,
+            rewardName: data['event']['reward']['title'],
+            rewardCost: data['event']['reward']['cost']);
+        yield AppendDeltaEvent(model);
```

Comment:
Should this be an UpdateDeltaEvent and modify the previous event?

Reply:
I don't think it would need to update the previous one for regular scenarios since I'm just grabbing fulfilled events, maybe if the redemption could change from fulfilled to canceled.

But we just need some data to really see how these two events behave.

Reply:
I think it [might](https://help.twitch.tv/s/article/channel-points-guide?language=en_US) also change depending on how each streamer has it set up. I think for example some of the ones cat6 might have (sprint, cuss out loud) might never end up being fulfilled. And then I'd say automated ones should end up being fulfilled (like Pelotoms)